Common subdirectories: orig/dwm-4.7/debian/config and work/dwm-4.7/debian/config
Common subdirectories: orig/dwm-4.7/debian/desktop and work/dwm-4.7/debian/desktop
diff -Nu orig/dwm-4.7/debian/dwm.install work/dwm-4.7/debian/dwm.install
--- orig/dwm-4.7/debian/dwm.install	1969-12-31 21:00:00.000000000 -0300
+++ work/dwm-4.7/debian/dwm.install	2008-01-03 23:07:20.000000000 -0200
@@ -0,0 +1,2 @@
+bin/* usr/bin
+man/* usr/share/man/man1
Common subdirectories: orig/dwm-4.7/debian/icon and work/dwm-4.7/debian/icon
Common subdirectories: orig/dwm-4.7/debian/lintian and work/dwm-4.7/debian/lintian
Common subdirectories: orig/dwm-4.7/debian/patches and work/dwm-4.7/debian/patches
diff -Nu orig/dwm-4.7/debian/rules work/dwm-4.7/debian/rules
--- orig/dwm-4.7/debian/rules	2008-01-04 13:21:07.000000000 -0200
+++ work/dwm-4.7/debian/rules	2008-01-04 16:43:12.000000000 -0200
@@ -17,12 +17,25 @@
 build-stamp: patch-stamp
 	dh_testdir
 
-	# Building package
-	CFLAGS="$(CFLAGS)" $(MAKE) PREFIX=/usr
+	# Building alternatives
+	mkdir bin
+	mkdir man
+	cp config.h debian/config/config.default.h
+
+	set -e; for ALTERNATIVE in debian/config/*; \
+	do \
+		ALTERNATIVE_NAME=`basename $$ALTERNATIVE | sed -e 's/config.//' -e 's/.h//'`; \
+		$(MAKE) clean; \
+		cp $$ALTERNATIVE config.h; \
+		CFLAGS="$(CFLAGS)" $(MAKE) PREFIX=/usr; \
+		mv dwm bin/dwm.$$ALTERNATIVE_NAME; \
+		ln -s dwm.1.gz man/dwm.$$ALTERNATIVE_NAME.1.gz; \
+	done
 
 	touch build-stamp
 
-clean:
+clean: clean-patched unpatch
+clean-patched:
 	dh_testdir
 	dh_testroot
 	rm -f build-stamp
@@ -30,9 +43,8 @@
 	# Cleaning package
 	[ ! -f Makefile ] || $(MAKE) clean
 	rm -f debian/postinst debian/prerm
-	-mv config.default.h config.h
-
-	dpatch deapply-all
+	-mv debian/config/config.default.h config.h
+	-rm -r bin man
 
 	dh_clean
 
@@ -41,27 +53,16 @@
 	dh_testroot
 	dh_clean -k
 	dh_installdirs
-
-	# Installing package
-	$(MAKE) install DESTDIR=$(CURDIR)/debian/dwm PREFIX=/usr
-	mv debian/dwm/usr/bin/dwm debian/dwm/usr/bin/dwm.default
-	ln -s dwm.1.gz debian/dwm/usr/share/man/man1/dwm.default.1.gz
-
-	# Building alternatives
-	cp config.h config.default.h
-
-	set -e; for ALTERNATIVE in debian/config/*; \
-	do \
-		$(MAKE) clean; \
-		cp $$ALTERNATIVE config.h; \
-		CFLAGS="$(CFLAGS)" $(MAKE) PREFIX=/usr; \
-		install -m 0755 dwm debian/dwm/usr/bin/dwm.`basename $$ALTERNATIVE | sed -e 's/config.//' -e 's/.h//'`; \
-		ln -s dwm.1.gz debian/dwm/usr/share/man/man1/dwm.`basename $$ALTERNATIVE | sed -e 's/config.//' -e 's/.h//'`.1.gz; \
-	done
+	dh_install
+	dh_installman dwm.1
 
 	# Configuring maintainer scripts
-	sed -e 's/@ALTERNATIVES@/web/' debian/postinst.in > debian/postinst; \
-	sed -e 's/@ALTERNATIVES@/web/' debian/prerm.in > debian/prerm; \
+	for ALTERNATIVE in bin/*; \
+	do \
+		ALTERNATIVES="$$ALTERNATIVES `basename $$ALTERNATIVE | sed -e 's/dwm.//'`"; \
+	done; \
+	sed -e "s/@ALTERNATIVES@/$$ALTERNATIVES/" debian/postinst.in > debian/postinst; \
+	sed -e "s/@ALTERNATIVES@/$$ALTERNATIVES/" debian/prerm.in > debian/prerm; \
 
 	# Installing desktop file
 	install -D -m 0644 debian/desktop/dwm debian/dwm/usr/share/xsessions/dwm.desktop
